domain/application/state/unit_test.go

```diff
@@ -1347,6 +1403,52 @@ WHERE uuid=?`, constraintUUID)
 	return constraintSpaces, constraintTags, constraintZones
 }
 
+func (s *unitStateSuite) addUnit(c *gc.C, unitName coreunit.Name, appUUID coreapplication.ID) coreunit.UUID {
+	unitUUID := coreunittesting.GenUnitUUID(c)
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		netNodeUUID := uuid.MustNewUUID().String()
+		_, err := tx.Exec(`
+INSERT INTO net_node (uuid)
+VALUES (?)
+`, netNodeUUID)
+		if err != nil {
+			return err
+		}
+
+		_, err = tx.Exec(`
+INSERT INTO unit (uuid, name, life_id, net_node_uuid, application_uuid, charm_uuid)
+SELECT ?, ?, ?, ?, uuid, charm_uuid
+FROM application
+WHERE uuid = ?
+`, unitUUID, unitName, 0 /* alive */, netNodeUUID, appUUID)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	return unitUUID
+}
+
+func (s *unitStateSuite) addMachineToUnit(c *gc.C, unitUUID coreunit.UUID) (coremachine.Name, coremachine.UUID) {
+	machineUUID := coremachinetesting.GenUUID(c)
+	machineName := coremachine.Name("0")
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.Exec(`
+INSERT INTO machine (uuid, name, life_id, net_node_uuid)
+SELECT ?, ?, ?, net_node_uuid
+FROM unit
+WHERE uuid = ?
+`, machineUUID, machineName, 0 /* alive */, unitUUID)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	return machineName, machineUUID
+}
```

Comment:
I would prefer this were done with `createApplication`, but that doesn't return the machine name or uuid.

This part probable needs some more thought from our side

Reply:
Actually, I see this is what has been done elsewhere already, so I think we should go ahead for now and come back to this later